### ISSUE TITLE
quick setup of meduim prioirty road 

### ISSUE THREAD

**Issue by kianzarrin** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#712]
![aubergines road](https://user-images.githubusercontent.com/1386719/73811819-70d60580-47d2-11ea-8674-1757924abbef.png)

## Introduction
In https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/pull/621#issuecomment-582545064 , https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/pull/621#issuecomment-583129735 , and  @aubergine10 requested an alternative high priority road (lets call it medium priority road) that does not allow pedestrian crossings but does allow going through blocked junctions (@aubergine10  what about far side turn?) .
I created [this uncommitted branch]( https://github.com/kianzarrin/Cities-Skylines-Traffic-Manager-President-Edition/tree/AltPriorityRoad) that uses alt key for the said medium priority road for demo.

The problem is the number of modifier keys is getting out of hand. I think too many modifier keys and state machines can confuse users.

So the question is what should be the UI that supports alternative priority roads for user?

related #557 #710 #691 #711

## Proposed solutions:

- Put it in road selection panel (#631)
- Create a GUI panel for priority road #711
- an in game  panel to select which  rules we want to be affected like the roundabout mod:
![Screenshot (413)](https://user-images.githubusercontent.com/26344691/74760907-cbb43600-5283-11ea-8781-875cb51ee0df.png)

**Comment by kianzarrin** [comment]
@aubergine10 In the medium priority road I assume you want:
1- going through blocked junction
2- pedestrian crossings on the main road

Do you want cars to be able to do left turn? (I think you should otherwise cars have to go round the city to turn left.)

**Comment by aubergine10** [comment]
Yes to 1 & 2. And yes for far-side turns. That should be sufficient to differentiate the medium vs. high priority road.

Maybe we should also default to medium priority if user has selected a 'small road' (or high priority for medium/large road)...?

I worry that if users go round splatting their roads with the high priority settings that their city will be a nightmare for pathfinder.

_Also, on the high priority one we already have, I assume that traffic lights will enable vehicles to reach the other half of the city? (I can't remember if I tested that yet but will check next time I'm in game)._

**Comment by kianzarrin** [comment]
> Also, on the high priority one we already have, I assume that traffic lights will enable vehicles to reach the other half of the city? (I can't remember if I tested that yet but will check next time I'm in game).

Nope!!!! see https://github.com/CitiesSkylinesMods/TMPE/wiki/High-priority-Road#warning-you-will-need-a-collector-road

So I am worried about that to but I don't know what to do about it. The user should limit the range of the high priority road to exclude one road at the beginning or the end. Maybe I can do a path-find my self to see if the car has to drive too far or not. if the path is too far away then I can pop-up a guide message (hey did you know with these new settings your car has to drive all the way around the city now? please give your cars more options and create a collector road).

But even if the cars are driving all the way across the city its still better than some nightmare setups that people have with too many junctions close together. At the very least the traffic can flow!
```
      |      |      |      |      |      |
======+======+======+======+======+======+======
      |      |      |      |      |      |       
```

**Comment by kianzarrin** [comment]
> Yes to 1 & 2. And yes for far-side turns

Got you!!!!!! this is in contradiction to #424 . If #454 is implemented and far side turns is enabled then cars should nomraly keep clear of intersection.

If #424 is implemented then there is no need for a medium priority road (as you described above) and you can simply use the yield/stop button.

**Comment by kianzarrin** [comment]
in any case #889 should fix this issue for the most part. Should I close this as duplicate of #889?

**Comment by aubergine10** [comment]
> Should I close this as duplicate of #889?

In #889 I'd ideally want to be able to define settings for high priority and medium priority separately, and then maybe have `Alt + Click` modifier to decide what clicking the priority road button does. In fact, in that situation the distiction would be 'normal vs. alternate' rather than 'high vs. low' - essentially I have two profiles to choose from, the normal profile (just normal click the button) or the alternate profile (Alt+Click button). The same could apply to roundabouts; eg. I could set up two profiles, one for big roundabouts, one for small roundabouts and choose to apply the alternate one with an Alt+Click.

**Comment by aubergine10** [comment]
Also, not sure why you referenced #454 "Trace in RELEASE build"? Did you mean #424 ?

**Comment by kianzarrin** [comment]
where do you setup the alternative profile though?
and if #424 is implemented you cannot have what you described so at least the issue has to change. I should change issue description.
But if we cannot setup enter blocked junction I think the alternative profile would not be so useful anymore. this also means in #889 I might not actually implement the alternative profile to simplify things. Instead I provide  a bulk edit button which is fully customizable.
